src/components/Dialog/Dialog.tsx

```diff
@@ -250,15 +250,21 @@ export default class Dialog extends Component<DialogProps, DialogState> {
       if (sources.length === 0) {
         throw noSourcesError();
       }
+      const invocationParams = this.props.sdk.parameters
+        .invocation as AppInvocationParameters;
+      const installationParams = this.props.sdk.parameters
+        .installation as AppInstallationParameters;
+
       // check if previously selected source exists
       // if it does, add it to state.
-      const previouslySelectedSource = (
-        this.props.sdk.parameters.invocation as AppInvocationParameters
-      )?.selectedImage?.selectedSource;
+      const previouslySelectedSource =
+        invocationParams?.selectedImage?.selectedSource?.id ||
+        installationParams.sourceID;
```

Comment:
suggestion (non-blocking): I almost called out that the comparison below was in invalid, but seeing what `previouslySelectedSource` actually is, I see that its a direct comparison.

I think we should rename this variable `previouslySelectedSourceId` and update where needed.

```suggestion
      const previouslySelectedSourceId =
        invocationParams?.selectedImage?.selectedSource?.id ||
        installationParams.sourceID;
```

Reply:
comment:

I can see why that's confusing 😅 , I can update this variable name to make this easier to grok 👍🏼 .